.github/workflows/l10n.yml

```diff
@@ -0,0 +1,1098 @@
+    runs-on: ${{ matrix.job.os }}
+    env:
+      SCCACHE_GHA_ENABLED: "true"
+      RUSTC_WRAPPER: "sccache"
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          - { os: ubuntu-latest  , features: "feat_os_unix" }
+          - { os: macos-latest   , features: "feat_os_macos" }
+          - { os: windows-latest , features: "feat_os_windows" }
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+    - uses: dtolnay/rust-toolchain@stable
+    - uses: taiki-e/install-action@nextest
+    - uses: Swatinem/rust-cache@v2
+    - name: Run sccache-cache
+      uses: mozilla-actions/sccache-action@v0.0.9
+    - name: Install/setup prerequisites
+      shell: bash
+      run: |
+        ## Install/setup prerequisites
+        case '${{ matrix.job.os }}' in
+          ubuntu-*)
+            # selinux headers needed for testing
+            sudo apt-get -y update ; sudo apt-get -y install libselinux1-dev
+          ;;
+          macos-*)
+            # needed for testing
+            brew install coreutils
+          ;;
+        esac
+    - name: Build with platform features
+      shell: bash
+      run: |
+        ## Build with platform-specific features to enable l10n functionality
+        cargo build --features ${{ matrix.job.features }}
+    - name: Test l10n functionality
+      shell: bash
+      run: |
+        ## Test l10n functionality
+        cargo test -p uucore locale
+        cargo test
+      env:
+        RUST_BACKTRACE: "1"
+
+  l10n_fluent_syntax:
+    name: L10n/Fluent Syntax Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.x'
+    - name: Install Mozilla Fluent Linter
+      shell: bash
+      run: |
+        ## Install Mozilla Fluent Linter
+        pip install moz-fluent-linter
+    - name: Find and validate Fluent files
+      shell: bash
+      run: |
+        ## Find and validate Fluent files with Mozilla Fluent Linter
+
+        # Check if any .ftl files exist
+        fluent_files=$(find . -name "*.ftl" -type f 2>/dev/null || true)
+
+        if [ -z "$fluent_files" ]; then
+          echo "::notice::No Fluent (.ftl) files found in the repository"
+          exit 0
+        fi
+
+        echo "Found Fluent files:"
+        echo "$fluent_files"
+
+        # Use Mozilla Fluent Linter for comprehensive validation
+        echo "Running Mozilla Fluent Linter..."
+
+        has_errors=false
+
+        while IFS= read -r file; do
+          echo "Checking $file with Mozilla Fluent Linter..."
+
+          # Run fluent-linter on each file
+          if ! moz-fluent-lint "$file"; then
+            echo "::error file=$file::Fluent syntax errors found in $file"
+            has_errors=true
+          else
+            echo "✓ Fluent syntax check passed for $file"
+          fi
```

Comment:
For consistency purposes I would remove the negation and swap the `if` and `else` blocks. Later in the file you use the `if` blocks for the "success" cases and the `else` blocks for the "failure" cases.